### PR TITLE
Move version-conversion code out of shared kubectl resource package

### PR DESCRIPTION
**Pull request by pwittrock**
```release-note
NONE
```

### CODE REVIEW

**Comment by pwittrock** [comment]
cc @liggitt FYI, doing this instead

File: pkg/kubectl/cmd/convert.go [hunk 1]
```diff
@@ -201,3 +202,99 @@ func (o *ConvertOptions) RunConvert() error {
...
+			return nil, err
+		}
+		objects = append(objects, converted)
+	}
+	return objects, nil
+}
+
+// TryConvert attempts to convert the given object to the provided versions in order. This function assumes
+// the object is in internal version.
+func TryConvert(converter runtime.ObjectConvertor, object runtime.Object, versions ...schema.GroupVersion) (runtime.Object, error) {
```
Comment (liggitt) [inline pkg/kubectl/cmd/convert.go]: would prefer not to export this from this package

**Comment by liggitt** [comment]
cc @deads2k @fabianofranz 

I don't object to removing conversion gradually. I think removing the pieces that require internal objects, like printers, are a better place to start, because they're likely to take the longest and be on  the critical path for completely getting rid of internal dependencies.

**Comment by apelisse** [comment]
Yeah looks good to me.

**Comment by pwittrock** [comment]
@liggitt LMK if this does not address your comment.

The motivation for this is to be able to move the resource builder code out of the main repo so that it can be shared by other projects.  Particularly, the [app-def working group](https://github.com/kubernetes/community/tree/master/wg-app-def) needs to use this code.  While 

I would like to remove the conversion pieces from kubectl altogether, it is more urgent that the kubectl infrastructure shared by plugins and other CLIs be made available in a shared location.

**Comment by liggitt** [comment]
only comment on this PR was unexporting the TryConvert

File: pkg/kubectl/cmd/convert.go (hunk 1 above)
Comment (monopole) [reply pkg/kubectl/cmd/convert.go]: agreed, looks like there's no reason to have it this public anymore.  Thanks for doing this Phil.

**Comment by monopole** [comment]
s/TryConvert/tryConvert/
and it /lgtm

**Pull request merged**